analyzers/src/SonarAnalyzer.CSharp/Rules/CheckArgumentException.cs

```diff
@@ -124,21 +124,25 @@ or ParenthesizedLambdaExpressionSyntax
                     or AccessorDeclarationSyntax
                     or BaseMethodDeclarationSyntax
                     or IndexerDeclarationSyntax
+                    or PropertyDeclarationSyntax
```

Comment:
I changed the logic a bit - because the node here is the first Ancestor that fulfills the criteria in lines 121 to 129 - so before I was getting the first ancestor record, which was wrong, (since the parameter we are checking might be in a method, that is in a class, that is in a record) and from our discussion, I understood that we don't really want that.